File: upload_photos.js
const fs = require('fs');
const path = require('path');
const Database = require('better-sqlite3');

const dbPath = "C:/Users/Admin/AppData/Roaming/school-management-system/school.db";
const photosDir = "C:/Users/Admin/Desktop/School Work/Students Photo";

const db = new Database(dbPath);

async function main() {
  const files = fs.readdirSync(photosDir);
  const updatedStudents = [];
  const notFound = [];
  const multipleFound = [];

  for (const file of files) {
    if (file.startsWith('file_') || file === 'v.png') continue;
    
    // Extract name (e.g., "Afreen.png" -> "Afreen")
    const ext = path.extname(file);
    const nameToMatch = path.basename(file, ext).trim();
    
    if (!nameToMatch) continue;

    // Search for student in database
    const queryName = `%${nameToMatch}%`;
    const rows = db.prepare(`SELECT * FROM students WHERE fullName LIKE ? COLLATE NOCASE`).all(queryName);

    if (rows.length === 1) {
      const student = rows[0];
      const filePath = path.join(photosDir, file);
      const fileData = fs.readFileSync(filePath);
      
      // Convert to base64
      let mimeType = 'image/jpeg';
      if (ext.toLowerCase() === '.png') mimeType = 'image/png';
      
      const base64Str = fileData.toString('base64');
      const dataUrl = `data:${mimeType};base64,${base64Str}`;

      // Update student
      db.prepare(`UPDATE students SET photo = ? WHERE id = ?`).run(dataUrl, student.id);
      
      updatedStudents.push(`${student.fullName} (Matched with ${file})`);
    } else if (rows.length === 0) {
      notFound.push(nameToMatch);
    } else {
      multipleFound.push(nameToMatch);
    }
  }

  console.log("=== SUCCESSFULLY ADDED ===");
  updatedStudents.forEach(s => console.log("- " + s));
  
  console.log("\n=== NOT FOUND IN DATABASE ===");
  notFound.forEach(s => console.log("- " + s));
  
  console.log("\n=== MULTIPLE MATCHES (Skipped) ===");
  multipleFound.forEach(s => console.log("- " + s));
  
  process.exit(0);
}

main().catch(err => {
  console.error(err);
  process.exit(1);
});

File: upload_photos.py
import os
import sqlite3
import base64

db_path = r"C:\Users\Admin\AppData\Roaming\school-management-system\school.db"
photos_dir = r"C:\Users\Admin\Desktop\School Work\Students Photo"

def main():
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()

    files = os.listdir(photos_dir)
    updated_students = []
    not_found = []
    multiple_found = []

    for file in files:
        if file.startswith("file_") or file == "v.png":
            continue

        name_to_match = os.path.splitext(file)[0].strip()
        if not name_to_match:
            continue

        # Search for student in database (case-insensitive)
        query_name = f"%{name_to_match}%"
        cursor.execute("SELECT id, fullName FROM students WHERE fullName LIKE ?", (query_name,))
        rows = cursor.fetchall()

        if len(rows) == 1:
            student_id, student_name = rows[0]
            file_path = os.path.join(photos_dir, file)
            
            with open(file_path, "rb") as image_file:
                encoded_string = base64.b64encode(image_file.read()).decode('utf-8')
            
            ext = os.path.splitext(file)[1].lower()
            mime_type = "image/png" if ext == ".png" else "image/jpeg"
            data_url = f"data:{mime_type};base64,{encoded_string}"

            cursor.execute("UPDATE students SET photo = ? WHERE id = ?", (data_url, student_id))
            updated_students.append(f"{student_name} (Matched with {file})")
        elif len(rows) == 0:
            not_found.append(name_to_match)
        else:
            multiple_found.append(name_to_match)

    conn.commit()
    conn.close()

    print("=== SUCCESSFULLY ADDED ===")
    for s in updated_students:
        print(f"- {s}")
    
    print("\n=== NOT FOUND IN DATABASE ===")
    for s in not_found:
        print(f"- {s}")
    
    print("\n=== MULTIPLE MATCHES (Skipped) ===")
    for s in multiple_found:
        print(f"- {s}")

if __name__ == "__main__":
    main()
